### PR TITLE
[CHAL-52] Refactor bit.dev Pin input component and make it compatible with safari 13.1.1

### PR DESCRIPTION
**Pull request by ekarthiek**
**The following must be included in every PR. Please check all items before submitting your request.
If an item can not be satisfied, please provide a brief explanation as to why the item is not included.**

- [x] Rebased from Master prior to creation of pull request
- [ ] Minimum 1 unit test that covers the component
- [x] All tests run and passed prior to creation of pull request
- [ ] Component added to bit.dev server
- [ ] New feature documentation added to bit.dev
- [x] Ran `yarn sync` to get most recent version of all components from bit.dev server
- [x] Sandbox example of your component added/updated to App.js

**Provide a brief description of your update:**
JIRA_LINK:https://campgladiator.atlassian.net/browse/CHAL-52
FIGMA_LINK: https://www.figma.com/file/c0C006ONITlaI0RKactNAI/Spring-Challenge-2021?node-id=904%3A6656

-  Fixed the pin input component and made it compatible with safari
- Updated the view icon as per the design

**Steps To Test:**
The pin input component seems to be working fine when we run in the development mode. For testing this update,
1. Build the Application using `yarn build`
2. Test the application build in safari browser and other browsers as well

### CODE REVIEW

File: src/components/molecules/PinInput/PinInput.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import mods from '../../../utilities/mods'
-import Icon from '../../atoms/Icon'
 import Button from '../../atoms/Button'
 import './PinInput.scss'
-
+import keyCodes from '../../../utilities/keycodes'
+import { CDN_URL } from '../../../utilities/path'
 class PinInput extends Component {
```
Comment (PrashantKIsHere) [inline src/components/molecules/PinInput/PinInput.js:8]: Add Line Gap Here

File: src/components/molecules/PinInput/PinInput.js [hunk 2]
```diff
@@ -34,64 +34,45 @@ class PinInput extends Component {
...
-    this.pinInputRef = null
-    this.setInputPartRefCallback = this.setInputPartRefCallback.bind(this)
-    this.onInputChangeHandler = this.onInputChangeHandler.bind(this)
-    this.viewInputData = this.viewInputData.bind(this)
+    this.pinInputRef = []
+  }
+
+  setRef = (ref) => {
+    this.pinInputRef.push(ref)
   }
```
Comment (PrashantKIsHere) [inline src/components/molecules/PinInput/PinInput.js:45]: Change this to

```
setRef = (ref) => this.pinInputRef.push(ref)
```

File: src/components/molecules/PinInput/PinInput.js [hunk 3]
```diff
@@ -34,64 +34,45 @@ class PinInput extends Component {
...
 
   convertNoOfInputToStateObj() {
     const { noOfInput, value } = this.props
     const updatedValue = value || ''
-    return Array.from(Array(noOfInput), (_, index) => index + 1).reduce(
```
Comment (PrashantKIsHere) [inline src/components/molecules/PinInput/PinInput.js:51]: Change the `convertNoOfInputToStateObj(){`

```
convertNoOfInputToStateObj = () => {
```

File: src/components/molecules/PinInput/PinInput.js [hunk 4]
```diff
@@ -34,64 +34,45 @@ class PinInput extends Component {
...
   getFullPinValues() {
     const { pinValue } = this.state
-    let fullPinValues = ''
-    Object.keys(pinValue).map((item) => {
-      fullPinValues += pinValue[item]
-      return item
-    })
-    return fullPinValues
+    return Object.values(pinValue).join('')
   }
```
Comment (PrashantKIsHere) [inline src/components/molecules/PinInput/PinInput.js:60]: Update this to Just

```
getFullPinValues = () => Object.values(this.state.pinValue).join('')
```

File: src/components/molecules/PinInput/PinInput.js [hunk 5]
```diff
@@ -34,64 +34,45 @@ class PinInput extends Component {
...
 
   handleChange(e) {
     const { id, value } = e.target
     const { pinValue } = this.state
-    this.setState(
```
Comment (PrashantKIsHere) [inline src/components/molecules/PinInput/PinInput.js:81]: Use Arrow Function Here:

```
handleChange = (e) => {
```

File: src/components/molecules/PinInput/PinInput.js [hunk 6]
```diff
@@ -34,64 +34,45 @@ class PinInput extends Component {
...
+        const fullPinValues = this.getFullPinValues()
+        const userPin = fullPinValues.length === noOfInput ? fullPinValues : ''
+        onPinChanged({
+          userPin,
+        })
+      }
+    })
   }
 
   focusOnPinPart(indexPos) {
```
Comment (PrashantKIsHere) [inline src/components/molecules/PinInput/PinInput.js:78]: Use Arrow Function Here

```
focusOnPinPart = (indexPos) => {
```

File: src/components/molecules/PinInput/PinInput.js [hunk 7]
```diff
@@ -101,7 +82,7 @@ class PinInput extends Component {
     }
   }
 
-  onInputChangeHandler(e, index) {
+  onInputChangeHandler(index, e) {
```
Comment (PrashantKIsHere) [inline src/components/molecules/PinInput/PinInput.js:85]: Use Arrow Function Herw:

```
onInputChangeHandler = (index, e) => {
```

Comment (PrashantKIsHere) [reply src/components/molecules/PinInput/PinInput.js:85]: Additionally the statement 

```
const step = value ? 1 : -1
```

can be moved inside the if block

File: src/components/molecules/PinInput/PinInput.js [hunk 8]
```diff
@@ -111,44 +92,49 @@ class PinInput extends Component {
...
+  onInputKeyDown = (index, e) => {
+    if (e.key === keyCodes.BACKSPACE) {
+      this.focusOnPinPart(index - 1)
+      e.target.value = ''
+      this.handleChange(e)
+      e.preventDefault()
+    }
+  }
+
   getPinInputPart() {
```
Comment (PrashantKIsHere) [inline src/components/molecules/PinInput/PinInput.js:109]: Use Arrow Function

```
getPinInputPart = () => {
```

File: src/components/molecules/PinInput/PinInput.js [hunk 9]
```diff
@@ -111,44 +92,49 @@ class PinInput extends Component {
...
               autoComplete="off"
               value={value}
-              pattern="[0-9]*"
+              pattern="^[0–9]$"
               required
               name={name}
               inputMode="numeric"
-              onChange={onInputChangeHandler}
+              onKeyDown={this.onInputKeyDown.bind(this, index)}
+              onChange={this.onInputChangeHandler.bind(this, index)}
```
Comment (PrashantKIsHere) [inline src/components/molecules/PinInput/PinInput.js:137]: Update this with

```
              onKeyDown={(e) => this.onInputKeyDown(index, e)}
              onChange={(e) => this.onInputChangeHandler(index, e)}
```

**Review by PrashantKIsHere — CHANGES_REQUESTED** [review]

**Review by PrashantKIsHere — APPROVED** [review]
Approved

**Review by Rajmohanrk — APPROVED** [review]

**Pull request merged**